packages/studio-plugin/src/createStudioStylingPlugin.ts

```diff
@@ -0,0 +1,49 @@
+import { PluginOption } from "vite";
+
+/**
+ * Adds the ?inline query parameter to all CSS imports
+ * unless they are CSS from Studio, which is determined
+ * from the path of its importer. This prevents user styling
+ * from affecting Studio UI.
+ */
+export default function createStudioStylingPlugin(): PluginOption {
+  return {
+    name: "yext-studio-style-plugin",
+    enforce: "pre",
+    async resolveId(id, importer) {
+      if (!importer) {
+        return;
+      }
+      if (isImportedByStudio(importer)) {
+        return;
+      }
+      if (isStyleFile(id)) {
+        const resolvedId = (
+          await this.resolve(id, importer, { skipSelf: true })
+        )?.id;
+        if (resolvedId) {
+          return addQueryParameter(resolvedId, "inline");
+        }
+        throw new Error(
+          "Styling file could not be resolved by yext-studio-style-plugin"
+        );
+      }
+    },
+  };
+}
+
+function isImportedByStudio(id: string) {
+  return !!(id.match(/.*@yext\/studio.*/) || id.match(/.*packages\/studio*/));
```

Comment:
I think assuming `/` is the path separator could be an issue on windows. have you tested that?

Reply:
will test the new unix conversion implementation on my windows laptop and will update!

Reply:
I realized by laptop is set up to use WSL which is just linux-- would it be OK if I run tests on the windows laptop in office monday?

Reply:
since this is being merged into a feature branch, I'm fine with that!